src/components/Leaderboard/QuestsModalContent/QuestsModalContent.tsx

```diff
@@ -0,0 +1,68 @@
+import { Truncate, Text, CheckMarkIcon } from '@tg-app/ui';
+import './QuestsModalContent.css';
+import { Video } from '@tg-app/api';
+import { useEffect, useState } from 'react';
+import { ActiveTab } from '~/App.tsx';
+import { Progress } from '@telegram-apps/telegram-ui';
+import { useBot } from '~/hooks';
+
+type Props = {
+  currentUser: { publicKey: string; score: number };
+  setActiveTab: (tab: ActiveTab) => void;
+};
+
+export const QuestsModalContent = ({ currentUser, setActiveTab }: Props) => {
+  const bot = useBot();
+
+  const [videos, setVideos] = useState<Video[]>([]);
+
+  useEffect(() => {
+    bot.getVideos().then(setVideos);
```

Comment:
Wouldn't be better to provide videos from Media component instead of double-fetching?